distributed/environments.py

```diff
@@ -0,0 +1,16 @@
+from __future__ import absolute_import, division, print_function
+
+
+class Environment(object):
+    def __init__(self, condition=None, setup=None):
+        self._condition = condition
+        self._setup = setup
```

Comment:
Perhaps default to the following:

```python
self.condition = condition or lambda: True
self.setup = setup or lambda: None
```

and then do away with the method definitions?